### PR TITLE
EOS-16553: Validate node count for setup configure

### PR DESCRIPTION
**Pull request by divya-vijay**
Earlier part of PR: https://github.com/Seagate/cortx-prvsnr/pull/828 
Raising this validation as a separate patch for easier review.

### CODE REVIEW

File: api/python/provisioner/commands/configure_setup.py [hunk 1]
```diff
@@ -91,126 +104,81 @@ def __attrs_post_init__(self):
...
-        if len(missing_params) > 0:
-            raise ValueError(f"Mandatory param missing {missing_params}")
+    validate_map = {f'{CLUSTER}': NetworkParamsValidation,
+                    f'{NODE}': NodeParamsValidation,
+                    f'{STORAGE_ENCLOSURE}': StorageNodeParamsValidation,
+                    f'{NODE_DEFAULT}': NodeNetworkParamsValidation,
+                    f'{STORAGE_DEFAULT}': StorageEnclosureParamsValidation}
+
+    @staticmethod
+    def _validate_node_count(num_of_nodes, content):
```
Comment (ypise) [inline api/python/provisioner/commands/configure_setup.py:118]: No validate method in config_setup. Move this to a dedicated validation package/module.
Also, don't use static method unless necessary.

File: api/python/provisioner/commands/validate_setup.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+import logging
```
Comment (ypise) [inline api/python/provisioner/commands/validate_setup.py:18]: Add a line describing what this file does.

File: api/python/provisioner/commands/validate_setup.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
+#
```
Comment (ypise) [inline api/python/provisioner/commands/validate_setup.py:1]: Just call this validate.
We'll park multiple classes under this header.
Also, change path to `api/python/provisioner/commands/validators/`
- node.py
- network.py
- storage.py
- release.py

File: api/python/provisioner/config.py [hunk 4]
```diff
@@ -166,6 +166,14 @@
 CONTROLLER_B = 'b'
 CONTROLLER_BOTH = 'both'
 
+# configure_setup related Params
```
Comment (ypise) [inline api/python/provisioner/config.py:169]: I don't see a need for these enums.
If absolutely necessary, localize them to the `configure_setup.py` file

File: api/python/provisioner/inputs.py [hunk 5]
```diff
@@ -467,54 +467,61 @@ class Release(ParamGroupInputBase):
...
+        _param_group, descr="Common Server configuration"
     )
-    data_public_interfaces: List = ParamGroupInputBase._attr_ib(
-        _param_group, descr="node data public network interfaces"
+    cluster_id: str = ParamGroupInputBase._attr_ib(
+        _param_group, descr="Cluster ID"
     )
-    data_private_interfaces: List = ParamGroupInputBase._attr_ib(
-        _param_group, descr="node data private network interfaces"
+    network_data_interfaces: List = ParamGroupInputBase._attr_ib(
```
Comment (ypise) [inline api/python/provisioner/inputs.py:523]: We need the demarcation for private and public interfaces.
Moreover, do we really need to prepend network here?
I believe it's handled in `configure_setup.py`.

Avoid major changes.

File: api/python/provisioner/validate_setup.py [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+import logging
```
Comment (ypise) [inline api/python/provisioner/validate_setup.py:20]: Add one-liner what this file does.
Why is this file duplicated?

File: api/python/provisioner/configure_setup.py [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+import configparser
```
Comment (ypise) [inline api/python/provisioner/configure_setup.py:20]: Add one-liner what this file does.
Also, why do we need a separate file?

**Review by ypise — CHANGES_REQUESTED** [review]

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

File: api/python/provisioner/commands/configure_setup.py (hunk 1 above)
Comment (divya-vijay) [reply api/python/provisioner/commands/configure_setup.py]: Done. Moved into `validate` file. 
Codacy does not allow standalone functions to stay without marking them as static, as per codacy rule R0201.

File: api/python/provisioner/commands/validate_setup.py (hunk 2 above)
Comment (divya-vijay) [reply api/python/provisioner/commands/validate_setup.py:20]: Done, added.

File: api/python/provisioner/commands/validate_setup.py (hunk 3 above)
Comment (divya-vijay) [reply api/python/provisioner/commands/validate_setup.py:1]: Looked like the PR was getting too long with configure and validation. So, to restrict the scope, have created a ticket and will be addressing module movement in it. https://jts.seagate.com/browse/EOS-18518

File: api/python/provisioner/config.py (hunk 4 above)
Comment (divya-vijay) [reply api/python/provisioner/config.py:189]: they were used in both `configure_setup` and `validate` files, so have kept it in `config.py` for now. 
If their scope is not getting extended, will separately localize them to both files and use.

File: api/python/provisioner/inputs.py (hunk 5 above)
Comment (divya-vijay) [reply api/python/provisioner/inputs.py]: Done. Kept the naming convention as such, and only made few changes for the new `config.ini` format.

File: api/python/provisioner/validate_setup.py (hunk 6 above)
Comment (divya-vijay) [reply api/python/provisioner/validate_setup.py]: This was some merge conflict that happened with that commit. Removed the duplicated file and updated code.

**Comment by divya-vijay** [comment]
This patch was addressed as part of PRs #1060 and #1071 and merged. Hence, closing this PR.

**Pull request closed without merging**